bundles/org.openhab.binding.revogi/src/main/java/org/openhab/binding/revogi/internal/RevogiSmartStripControlHandler.java

```diff
@@ -0,0 +1,172 @@
+    private final SwitchService switchService;
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    private @Nullable RevogiSmartStripControlConfiguration config;
+
+    public RevogiSmartStripControlHandler(Thing thing, StatusService statusService, SwitchService switchService) {
+        super(thing);
+        this.statusService = statusService;
+        this.switchService = switchService;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        switch (channelUID.getId()) {
+            case RevogiSmartStripControlBindingConstants.PLUG_1_SWITCH:
+                switchPlug(command, 1);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_2_SWITCH:
+                switchPlug(command, 2);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_3_SWITCH:
+                switchPlug(command, 3);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_4_SWITCH:
+                switchPlug(command, 4);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_5_SWITCH:
+                switchPlug(command, 5);
+                break;
+            case RevogiSmartStripControlBindingConstants.PLUG_6_SWITCH:
+                switchPlug(command, 6);
+                break;
+            case RevogiSmartStripControlBindingConstants.ALL_PLUGS:
+                switchPlug(command, 0);
+                break;
+            default:
+                logger.info("Sometring went wrong, we've got a message for {}", channelUID.getId());
+        }
+    }
+
+    private void switchPlug(Command command, int port) {
+        if (config == null) {
+            logger.warn("No config available, config object was null");
+            return;
+        }
+        if (command instanceof OnOffType) {
+            int state = convertOnOffTypeToState(command);
+            switchService.switchPort(config.getSerialNumber(), config.ipAddress, port, state);
+        }
+    }
+
+    private int convertOnOffTypeToState(Command command) {
+        if (command == OnOffType.ON) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(RevogiSmartStripControlConfiguration.class);
+        updateStatus(ThingStatus.UNKNOWN);
+
+        // Example for background initialization:
+        scheduler.execute(this::updateStripInformation);
+        Runnable runnable = RevogiSmartStripControlHandler.this::updateStripInformation;
+
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(runnable, 0, config.getPollInterval(), TimeUnit.SECONDS);
+        }
+
+        logger.debug("Finished initializing!");
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        if (pollingJob != null && !pollingJob.isCancelled()) {
```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (pollingJob != null) {
```